pallets/randomness/src/benchmarks.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+//! Benchmarking
+use crate::{BalanceOf, Call, Config, Pallet, RandomnessResults, Request, RequestType};
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite, Zero};
+use frame_support::traits::{Currency, Get};
+use frame_system::RawOrigin;
+use pallet_evm::AddressMapping;
+use sp_core::{H160, H256};
+use sp_runtime::traits::One;
+
+/// Create a funded user from the input
+fn fund_user<T: Config>(user: H160, fee: BalanceOf<T>) {
+	let total_minted = fee + <<T as Config>::Deposit as Get<BalanceOf<T>>>::get();
+	T::Currency::make_free_balance_be(&T::AddressMapping::into_account_id(user), total_minted);
+	T::Currency::issue(total_minted);
+}
+
+benchmarks! {
+	set_babe_randomness_results {}: _(RawOrigin::None)
+	verify { }
+
+	// set_output {
+	// 	// needs to be 2nd block to reflect expected costs every block
+	// 	crate::vrf::set_input::<T>();
+	// 	crate::vrf::set_output::<T>();
+	// 	crate::vrf::set_input::<T>();
+	// }: {
+	// 	// TODO: how to mock digests to insert Vrf PreDigest into digests
+	// 	let mut digests = frame_system::Pallet::<T>::digests().unwrap();
+	// 	// next push the expected PreDigest into the digests
+	// 	crate::vrf::set_output::<T>();
+	// }
+	// verify {}
```

Comment:
The current hardcoded weight returned for `set_output` is `T::DbWeight::get().read // TODO: update weight`. This must be changed before merging this PR (either by finishing this benchmark or setting that value to something conservatively high, either is better than the current code in `pallet-randomness/src/vrf.rs`)

Reply:
I changed it to a more accurate hardcoded value and made a follow up jira ticket [MOON-1804]